pyop2/compilation.py

```diff
@@ -81,23 +104,18 @@ def sniff_compiler_version(cc):
     else:
         compiler = "unknown"
 
-    ver = version.LooseVersion("unknown")
+    ver = UnknownVersion()
     if compiler == "gcc":
         try:
             ver = subprocess.check_output([cc, "-dumpversion"],
                                           stderr=subprocess.DEVNULL).decode("utf-8")
-            try:
-                ver = version.StrictVersion(ver.strip())
-            except ValueError:
-                # A sole digit, e.g. 7, results in a ValueError, so
-                # append a "do-nothing, but make it work" string.
-                ver = version.StrictVersion(ver.strip() + ".0")
-            if compiler == "gcc" and ver >= version.StrictVersion("7.0"):
+            ver = Version(ver.strip())
```

Comment:
Have you checked that a `ValueError` is no longer thrown for sole digits?

Reply:
Yes
```python
>>> from packaging.version import Version
>>> v = Version('3')
>>> print(v)
3
>>> v < Version('3.4.5')
True
```